etna/datasets/tsdataset.py

```diff
@@ -113,6 +113,7 @@ def __init__(self, df: pd.DataFrame, freq: str, df_exog: Optional[pd.DataFrame]
             self.df = self._merge_exog(self.df)
 
         self.transforms = None
+        self._update_regressors()
 
     def transform(self, transforms: Iterable["Transform"]):
         """Apply given transform to the data."""
```

Comment:
looks like we need to update `self._regressors` in `transform` too because this method creates new columns